client/src/document/editthispage.tsx

```diff
@@ -14,12 +14,13 @@ export function EditThisPage({ source }) {
         }
       }, 3000);
     }
+    return () => {
+      dismounted = true;
```

Comment:
This is an unrelated fix but worth doing. Anything async in an effect should always do this stuff. Actually, I had started (see the `let dismounted = false` a couple of lines above).

Reply:
Didn't figure out how to create a multi-line suggestion, so here it is in comment form. Canceling the timeout seems like the more sensible option here:

```jsx
  useEffect(() => {
    let timeoutId;
    if (opening) {
      timeoutId = setTimeout(() => {
        setOpening(false);
      }, 3000);
    }
    return () => {
      clearTimeout(timeoutId);
    };
  });
```

Reply:
done. Thanks!